tests/bats/helpers.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+#
+#  SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#  SPDX-License-Identifier: Apache-2.0
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+# Events accumulate over time, so for certainty it's best to use a unique pod
+# name. Right now, this inspects an entire line which includes REASON, MESSAGE,
+# and OBJECT, so choose the needle (grepped for) precisely enough.
+# Example: wait_for_pod_event pod/testpod-ls09x FailedPrepareDynamicResources 60
+wait_for_pod_event() {
```

Comment:
As far as I could find out, this cannot be achieved with `kubectl wait ...`. Hence, manual work :roller_coaster:.